(r.project) add bin directory to PATH

R.project doesnt add `bin` directory to PATH.

I am doing the following to find it and add it to the PATH, which should work as a drop in.  However as this package has the version already in the .ps1, it could make the version a variable and skip the find step.

```ps1
function Add-R-to-PATH
{
  # $list = Get-ChildItem -Directory 'C:\Program Files\R' | Out-String
  # Write-Host $list

  Get-ChildItem -Directory 'C:\Program Files\R' | % {
    $R_ROOT = $_.FullName

    Write-Host 'Setting R_ROOT='$R_ROOT

    Set-ItemProperty -path 'HKCU:\Environment' -name 'R_ROOT' -value $R_ROOT

    $R_BIN = ($R_ROOT + '\bin')

    Install-ChocolateyPath -PathToInstall $R_BIN
  }
  if ($R_ROOT) {
    return $R_ROOT
  }
  throw ('R not found in ' + $list)
}
```

Also I am experiencing that newly installed packages are not in the R libs path either, so it might also need to be added to `.Rprofile` -- still investigating that one.

The default repo can be added to `$R_ROOT/etc/Rprofile.site` with
```R
local({r <- getOption("repos")
       r["CRAN"] <- "http://cran.r-project.org"
       options(repos=r)})
```

The next problem is the compilers are not setup, resulting in errors.

```
The downloaded binary packages are in
	C:\Users\appveyor\AppData\Local\Temp\RtmpUVsoEq\downloaded_packages
installing the source package 'markdown'
trying URL 'http://cran.us.r-project.org/src/contrib/markdown_1.0.tar.gz'
Content type 'application/x-gzip' length 80843 bytes (78 KB)
==================================================
downloaded 78 KB
Warning in strptime(xx, f, tz = tz) :
  unable to identify current timezone 'C':
please set environment variable 'TZ'
* installing *source* package 'markdown' ...
** package 'markdown' successfully unpacked and MD5 sums checked
** using staged installation
** libs
*** arch - i386
c:/Rtools/mingw_32/bin/gcc  -I"C:/PROGRA~1/R/R-36~1.0/include" -DNDEBUG          -O3 -Wall  -std=gnu99 -mtune=generic -c Rbase64.c -o Rbase64.o
/bin/sh: c:/Rtools/mingw_32/bin/gcc: No such file or directory
make: *** [C:/PROGRA~1/R/R-36~1.0/etc/i386/Makeconf:208: Rbase64.o] Error 127
ERROR: compilation failed for package 'markdown'
* removing 'C:/Program Files/R/R-3.6.0/library/markdown'
```

--- a/automatic/r.project/Readme.md
+++ b/automatic/r.project/Readme.md
@@ -30,7 +30,8 @@ R has its own LaTeX-like documentation format, which is used to supply comprehen
 
 ## Package Parameters
 
-- `/UseInf:` - Change the inno setup configuration file to use/save when installing
+- `/UseInf:` - Change the inno setup configuration file to use/save when installing.
+- `/AddToPath` - Add the R.Project bin directory to the `PATH` environment variable.
 
 **EXAMPLE**
 `choco install r.project --params "'/UseInf:C:\r.project.inf'"`

--- a/automatic/r.project/r.project.nuspec
+++ b/automatic/r.project/r.project.nuspec
@@ -58,9 +58,13 @@ R has its own LaTeX-like documentation format, which is used to supply comprehen
 [Software Changelog](https://cloud.r-project.org/doc/manuals/r-release/NEWS.html)
 [Package Changelog](https://github.com/AdmiringWorm/chocolatey-packages/blob/master/automatic/r.project/Changelog.md)
     </releaseNotes>
+    <dependencies>
+      <dependency id="chocolatey-core.extension" version="1.3.3" />
+    </dependencies>
   </metadata>
   <files>
     <file src="legal\**" target="legal" />
     <file src="tools\**" target="tools" />
+    <file src="..\cmder\tools\Uninstall-ChocolateyPath.psm1" target="tools\Uninstall-ChocolateyPath.psm1" />
   </files>
 </package>

--- a/automatic/r.project/tools/chocolateyBeforeModify.ps1
+++ b/automatic/r.project/tools/chocolateyBeforeModify.ps1
@@ -1,0 +1,9 @@
+﻿$installDir = Get-AppInstallLocation ''
+
+if ($installDir) {
+  $toolsDir = Split-Path -Parent $MyInvocation.MyCommand.Definition
+  Import-Module "$toolsDir\Uninstall-ChocolateyPath.psm1"
+  # We want to remove the R.Project bin directory from PATH before upgrading,
+  # In case the install directory have changed.
+  Uninstall-ChocolateyPath "$installDir\bin"
+}

--- a/automatic/r.project/tools/chocolateyInstall.ps1
+++ b/automatic/r.project/tools/chocolateyInstall.ps1
@@ -23,3 +23,14 @@ if ($pp.UseInf) {
 }
 
 Install-ChocolateyInstallPackage @packageArgs
+
+if ($pp.AddToPath) {
+  # Get the location of R.Project install directory
+  $installDir = Get-AppInstallLocation $packageArgs["softwareName"]
+  if ($installDir -and (Test-Path "$installDir\bin")) {
+    Install-ChocolateyPath "$installDir\bin"
+  } else {
+    Write-Warning "Unable to detect installation directory, or directory do not have a bin directory."
+    Write-Warning "Skipping adding bin directory to PATH..."
+  }
+}

--- a/automatic/r.project/update.ps1
+++ b/automatic/r.project/update.ps1
@@ -18,7 +18,9 @@ function global:au_SearchReplace {
       "(?i)(checksum type:).*"   = "`${1} $($Latest.ChecksumType32)"
       "(?i)(checksum:).*"        = "`${1} $($Latest.Checksum32)"
     }
-
+    ".\tools\chocolateyBeforeModify.ps1" = @{
+      "(?i)(Get-AppInstallLocation\s*)'.*'" = "`${1}'$softwareName'"
+    }
     ".\tools\chocolateyInstall.ps1" = @{
       "(?i)(`"`[$]toolsDir\\).*`""        = "`${1}$($Latest.FileName32)`""
       "(?i)(^\s*softwareName\s*=\s*)'.*'" = "`${1}'$softwareName'"

